compiler/rustc_passes/src/loops.rs

```diff
@@ -246,12 +316,28 @@ impl<'a, 'hir> CheckLoopVisitor<'a, 'hir> {
         cf_type: &str,
     ) -> bool {
         if !span.is_desugaring(DesugaringKind::QuestionMark)
-            && self.cx == LabeledBlock
+            && *self.cx_stack.last().unwrap() == LabeledBlock
```

Comment:
Isn't this the same as the following?

```suggestion
            && self.cx_stack.last() == Some(LabeledBlock)
```

`cx_stack` should never be empty, but I find that stray `unwrap()`s tend to bite us later down the line.

Reply:
Done. Thank you.